app/registry.hs

```diff
@@ -449,13 +448,7 @@ decodeBech32
 decodeBech32 desc decoder text =
   case decoder text of
     Left e -> fail $ desc <> " decoding error for '"<>unpack text<>"':\n" <> show e
-    Right (humanPart, dataPart) -> pure $ Bech32 humanPart dataPart unprefixed
-      where
-        unprefixed :: Text
-        unprefixed = case Text.splitOn "_" text of
-                       []  -> error "Invariant failed:  Text.splitOn returned empty list."
-                       _:[] -> error "Invariant failed:  decoder succeeded on an unprefixed input text."
-                       _:x:_ -> x
+    Right (humanPart, dataPart) -> pure $ Bech32 humanPart dataPart
```

Comment:
Ok, this is simpler, but less user-pretty -- and I can see how simpler can be better in this setting.

cc @KtorZ

Reply:
Well, the validation was actually _wrong_ to some extend. It was removing a `prefix` based on the underscore symbol (`case Text.splitOn "_" text of`) however, bech32 prefixes are actually defined by the `1` separator, so in the case of `ed25519_pk`, this is the whole prefix! 

The way the validation now works is still rather userfriendly IMO. If you provide an "invalid" bech32 prefix for a particular data, it fails with a message as follows:

```
option --pledge-address: Pledge address has an unexpected human-readable part ( "ed25519_sk" ). Expected prefix is: "addr"
```